tests/framework/common_apps.go

```diff
@@ -277,3 +323,88 @@ func (td *OsmTestData) SimpleDeploymentApp(def SimpleDeploymentAppDef) (corev1.S
 
 	return serviceAccountDefinition, deploymentDefinition, serviceDefinition
 }
+
+// Handlers to  control plane apps below
+
+// GetGrafanaPodHandle forwards a grafana pod and returns a handler pointing to the local forwarded resource
+func (td *OsmTestData) GetGrafanaPodHandle(ns string, grafanaPodName string, port uint16) (*Grafana, error) {
+	portForwarder, err := k8s.NewPortForwarder(td.RestConfig, td.Client, grafanaPodName, ns, port, port)
+	if err != nil {
+		return nil, errors.Errorf("Error setting up port forwarding: %s", err)
+	}
+
+	err = portForwarder.Start(func(pf *k8s.PortForwarder) error {
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Errorf("Could not start forwarding: %s", err)
+	}
+
+	return &Grafana{
+		Schema:   "http",
+		Hostname: "localhost",
+		Port:     port,
+		User:     "admin", // default value of grafana deployment
+		Password: "admin", // default value of grafana deployment
+	}, nil
+}
+
+// GetPrometheusPodHandle forwards a prometheus pod and returns a handler pointing to the local forwarded resource
+func (td *OsmTestData) GetPrometheusPodHandle(ns string, prometheusPodName string, port uint16) (*Prometheus, error) {
+	portForwarder, err := k8s.NewPortForwarder(td.RestConfig, td.Client, prometheusPodName, ns, port, port)
+	if err != nil {
+		return nil, errors.Errorf("Error setting up port forwarding: %s", err)
+	}
+
+	err = portForwarder.Start(func(pf *k8s.PortForwarder) error {
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Errorf("Could not start forwarding: %s", err)
+	}
+
+	client, err := api.NewClient(api.Config{
+		Address: fmt.Sprintf("http://localhost:%d", port),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	v1api := v1.NewAPI(client)
+
+	return &Prometheus{
+		Client: client,
+		API:    v1api,
+	}, nil
+}
+
+// GetOSMPrometheusAndGrafanaHandles convenience wrapper, will get Prometheus and Grafana instances regularly deployed
+// by OSM installation in test OsmNamespace, and return a handle for each
+func (td *OsmTestData) GetOSMPrometheusAndGrafanaHandles() (*Prometheus, *Grafana, error) {
+	prometheusPod, err := Td.GetPodsForAppLabel(Td.OsmNamespace, metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app": OsmPrometheusAppLabel,
+		},
+	})
+	if err != nil || len(prometheusPod) == 0 {
+		return nil, nil, errors.Errorf("Error getting Prometheus pods: %v (prom pods len: %d)", err, len(prometheusPod))
+	}
+	pHandle, err := Td.GetPrometheusPodHandle(prometheusPod[0].Namespace, prometheusPod[0].Name, DefaultOsmPrometheusPort)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
Would it make sense to include this chunk in `GetPrometheusPodHandle` so that function doesn't need any arguments (and likewise for Grafana)? I could see using Prometheus in a test without Grafana so that would eliminate the need to duplicate this chunk in the test.

Reply:
good point.
One of the things I intended was to keep the semantics of OSM-specific things (fetching the pod with our label) in a separate function from the one that generically allows you to forward any prometheus instance.

Would at least split it into two different functions so they are independant, would that suffice?

Reply:
Yeah that makes sense